### PR TITLE
PD-1287: Textarea component accepts forward ref

### PR DESCRIPTION
**Pull request by brookescarlett**
## ✍️ Proposed changes

Upon request from DevHub

🎟 _Jira ticket:_ [PD-1287](https://jira.mongodb.org/browse/PD-1287)

## 🛠 Types of changes

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

### CODE REVIEW

File: packages/text-area/src/TextArea.tsx [hunk 1]
```diff
@@ -122,10 +123,10 @@ const colorSets: Record<Mode, ColorSets> = {
   },
 };
 
-type BaseTextAreaProps = HTMLElementProps<'textarea', never> & {
+type BaseTextAreaProps = HTMLElementProps<'textarea', HTMLTextAreaElement> & {
   id?: string;
   darkMode?: boolean;
-  label: string;
+  label: string | null;
```
Comment (DesignerDave) [inline packages/text-area/src/TextArea.tsx:129]: Can we be consistent with TextInput and make this optional as well?

File: packages/text-area/src/TextArea.tsx [hunk 2]
```diff
@@ -211,4 +218,17 @@ export default function TextArea({
...
   );
-}
+});
+
+TextArea.displayName = 'TextArea';
+
+TextArea.propTypes = {
+  id: PropTypes.string,
+  darkMode: PropTypes.bool,
+  label: PropTypes.string,
```
Comment (DesignerDave) [inline packages/text-area/src/TextArea.tsx:228]: This needs an update now, right?

**Review by DesignerDave — CHANGES_REQUESTED** [review]
Couple small things, but otherwise looks good!

File: packages/text-area/src/TextArea.tsx (hunk 1 above)
Comment (brookescarlett) [reply packages/text-area/src/TextArea.tsx:129]: yep

**Review by DesignerDave — APPROVED** [review]

**Pull request merged**